### DISCUSSION TITLE
Does Spark UI support server-side rendering (SSR)?

### DISCUSSION THREAD

**Discussion by halljus** [adevinta/spark#2323] [Q&A]
Hi all.

We're evaluating Spark for use in our Remix projects which are server-side rendered and noticed Spark components (or at least, certain ones like Combobox) have some issues with SSR. For example, we get the following warning in the server console:

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.`

And if we try to render certain things, we get hydration errors on the client as ids don't match what was rendered on the server, etc. Are these bugs or does Spark not support SSR?

Thanks.

**Comment by Powerplex** [comment]
Needs to be investigated. Leboncoin uses Spark with SSR in production and has no warning. Maybe a different stack/setup ?
